test/common/object_pool_test.cpp

```diff
@@ -34,6 +34,46 @@ TEST(ObjectPoolTests, SimpleReuseTest) {
   }
 }
 
+// Allocate more memory space than the object pool and expect exceptions
+// NOLINTNEXTLINE
+TEST(ObjectPoolTests, ExceedLimitTest) {
+  const uint32_t repeat = 1;
+  const uint64_t size_limit = 10;
+  const uint64_t reuse_limit = 10;
+  for (uint32_t iter = 0; iter < repeat; iter++) {
+    common::ObjectPool<uint32_t> tested(size_limit, reuse_limit);
+
+    // Get 11 objects
+    std::vector<uint32_t *> objects;
+    for (uint32_t i = 1; i <= 11; i++) {
+      uint32_t *cur_ptr = nullptr;
+      try {
+        cur_ptr = tested.Get();
+        if (i == 11) {
+          // free memory before we fail
+          tested.Release(cur_ptr);
+          for (auto &ptr : objects) tested.Release(ptr);
+          FAIL() << "Expect std::length_error.";
+        }
+        objects.push_back(cur_ptr);
+      } catch (const std::length_error &e) {
+        // got length error
+        if (i <= 10) {
+          // free memory before we fail
+          for (auto &ptr : objects) tested.Release(ptr);
+          FAIL() << "Unexpected std::length_error";
+        }
+      } catch (...) {
+        // free memory before we fail
+        for (auto &ptr : objects) tested.Release(ptr);
+        FAIL() << "Unexpected exceptions";
```

Comment:
Is there an easy way to print what exception actually got raised? If not, the test is small enough where this is probably easy to decipher.

Reply:
To my understanding, c++ allows you to throw any type of exceptions (int, char*, std::exception ...). I can't think of a way to catch any type. 

(Maybe boost::any can do the job, but I don't think that's necessary.)

Reply:
There should be EXPECT_THROW or something like that in gtest that you can use instead of this